packages/autorest.python/autorest/codegen/serializers/builder_serializer.py

```diff
@@ -949,11 +958,7 @@ def _call_request_builder_helper(  # pylint: disable=too-many-statements
         if builder.parameters.has_body and builder.parameters.body_parameter.flattened:
             # unflatten before passing to request builder as well
             retval.extend(_serialize_flattened_body(builder.parameters.body_parameter))
-        if (
-            builder.parameters.has_body
-            and builder.parameters.body_parameter.has_json_model_type
-            and any(p.in_flattened_body for p in builder.parameters.parameters)
-        ):
+        if is_json_model_type(builder.parameters):
```

Comment:
`is_json_model_type` is an unclear name. I think it should be `has_json_model_type` instead